client/starwhale/api/_impl/data_store.py

```diff
@@ -710,33 +868,116 @@ def dump(self) -> None:
 
 
 class RemoteDataStore:
-    def put(
-        self, table_name: str, schema: TableSchema, records: List[Dict[str, Any]]
+    def __init__(self, instance_uri: str) -> None:
+        self.instance_uri = instance_uri
+        self.token = os.getenv("SW_TOKEN")
+        if self.token is None:
+            raise RuntimeError("SW_TOKEN is not found in environment")
+
+    def update_table(
+        self,
+        table_name: str,
+        schema: TableSchema,
+        records: List[Dict[str, Any]],
     ) -> None:
-        ...
+        data: Dict[str, Any] = {"tableName": table_name}
+        schema_data: Dict[str, Any] = {
+            "keyColumn": schema.key_column,
+            "columnSchemaList": [
+                {"name": col.name, "type": str(col.type)}
+                for col in schema.columns.values()
+            ],
+        }
+        data["tableSchemaDesc"] = schema_data
+        if records is not None:
+            encoded: List[Dict[str, List[Dict[str, Optional[str]]]]] = []
+            for record in records:
+                r: List[Dict[str, Optional[str]]] = []
+                for k, v in record.items():
+                    if k == "-":
+                        r.append({"key": "-", "value": "1"})
+                    else:
+                        r.append({"key": k, "value": schema.columns[k].type.encode(v)})
+                encoded.append({"values": r})
+            data["records"] = encoded
+
+        assert self.token is not None
+        requests.post(
```

Comment:
add `timeout` for requests.post?   the default timeout in the request is the unlimited value.